### ISSUE TITLE
"It is diffcult or impossible to do obsure(very unique) things in Keras".  Is it?

### ISSUE THREAD

**Issue by subtleIsTheLord** [keras-team/keras#9308]
"It is diffcult or impossible to do obsure(very unique) things in Keras", this is what an expert in out department has suggested me. He advised me to switch to TF.  We are working on a Deep learning Project, which might lead us to propose our own CNN architecture and very different way of back-propagation, where loss function will be calculated at different layers simultaneously. We will be using CNNs inspired architecture for reconstruction and denoising of images. I am well familiar with CNN for classification in Keras, therefore I would obviously try to stay with Keras. Also learning curve of TF is little higher compare to KERAS. SO my question is what do u think of that guys statement  "It is diffcult or impossible to do obsure(very unique) things in Keras"? Has anybody experienced it?

**Comment by stoksc** [comment]
Looking at the variety of projects out there featuring Keras, I would tend to disagree with this. Also, don't forget Keras supports defining [custom layers](https://github.com/keras-team/keras/blob/master/examples/antirectifier.py).

**Comment by Carl-Granstrom** [comment]
Disclaimer: I'm just a beginner trying to learn more, so I might be way off on this one.

Well, if you use Keras on top of TF then what's the problem? You can use everything in TF and use Keras at the same time, no?

I'm no expert, far from it as a matter of fact, but it seems to me that it ought to be easier to modify an existing framework than to write everything you need "from scratch" in TF? Having different gradients from different loss functions that flow into a NN at different layers seems like something that Keras can handle.

**Comment by subtleIsTheLord** [comment]
Well, if you use Keras on top of TF then what's the problem? You can use everything in TF and use Keras at the same time, no?

The problem is I am novice to TF. And as said learning curve of TF is higher, so nothing but time.

Yes well said keras and TF can be easily mixed. Well known to me.
